packit/local_project.py

```diff
@@ -135,12 +137,16 @@ def _is_url(self, path_or_url):
             res = requests.head(path_or_url)
             if res.ok:
                 return True
-            else:
-                logger.warning("path_or_url is nor directory nor url")
+            logger.warning("path_or_url is nor directory nor url")
+        except requests.exceptions.InvalidSchema:
+            if re.match('^git@[\w\.]+[\w]\.[\w]+:[\w\-]+[\w]/[\w\.\:/\-]+(.git)?$', path_or_url):
```

Comment:
isn't possible to use `urllib module` at all here 
eg: ` urllib.parse.urlparse("git+ssh://example.com/some.git")`
produces
```
ParseResult(scheme='git-ssh', netloc='example.com', path='/some.git', params='', query='', fragment='')
```
and in case there is some prefix eg. `git@` or something, just remove this prefix and then use `urllib` again

Reply:
Honzo, that's definitely a good suggestion, but sadly it does not work for the ssh url github uses:
```python
$ ipython3

In [1]: from urllib.parse import urlparse

In [2]: urlparse("git@github.com:lachmanfrantisek/packit.git")
Out[2]: ParseResult(scheme='', netloc='', path='git@github.com:lachmanfrantisek/packit.git', params='', query='', fragment='')
```
So I guess that we still need to go for the regex nevertheless.

Reply:
Well, Honza already suggested removing the `git@`, but that still not works

```python
urlparse("github.com:lachmanfrantisek/packit.git")
ParseResult(scheme='github.com', netloc='', path='lachmanfrantisek/packit.git', params='', query='', fragment='')
```

Maybe replacing `git@` with `git+ssh://` ?

Reply:
yep, something like that, if we know the transformation, then we can do that.
replace `git@` by `git+ssh://` and first `:` by `/`
so code should be like:
```
url = "git@github.com:lachmanfrantisek/packit.git"

if urlparse(url).scheme:
   url_output = urlparse(url)
if url.startswith("git@"):
   url_output = urlparse(url.replace(":", "/", 1).replace("git@", "git+ssh://", 1))
```

Reply:
@jscotka looks good; @lbarcziova could you please incorporate the suggestion provided by Honza into the code? Thank you.

This code is also an excellent candidate for unit testing: are you up for the challenge or should we handle the integration ourselves?

Reply:
I will try to incorporate Honzas idea and make unit tests. I will submit as soon as possible

Reply:
I have another question. Can I understand your suggestion as the only checking of validity? Because in the case url = "git@github.com:lachmanfrantisekpackit.git" (invalid url because the / between user and repo is missing), 
`ParseResult(scheme='git+ssh', netloc='github.com', path='/lachmanfrantisekpackit.git', params='', query='', fragment='')`
this checking will pass. Or should I use regex for parsing the path in the ParseResult?

Reply:
I would start with just checking if it is ssh URL format. Later in code (for example, when we will try to clone the repo), we will find out that URL does not exist.

Also, it would be nice if this method has also docstring, so everyone knows what kind of validation it provides.

Reply:
+1 to @rpitonak's suggestion: we only need to know if the input is the ssh-style URL